src/NexusMods.DataModel/Interprocess/SqliteIPC.cs

```diff
@@ -207,14 +207,20 @@ private long ProcessMessages(long lastId)
                 var size = reader.GetBytes(2, 0, null, 0, 0);
                 var bytes = new byte[size];
                 reader.GetBytes(2, 0, bytes, 0, bytes.Length);
-                _subject.OnNext((queue, bytes));
+
+                _updates.Push((queue, bytes));
             }
         }
         catch (Exception ex)
         {
             _logger.LogError(ex, "Failed to process messages after {LastId}", lastId);
         }
 
+        while (_updates.TryPop(out var update))
+        {
+            _subject.OnNext(update);
```

Comment:
If I'm not mistaken this is effectively reversing the order of updates. Was the code already processing from newer to older and so this makes it right or will observers be notified of newer stuff first this way?

Reply:
Good catch, I forgot about LIFO/FIFO for a moment.